GATEWAY/src/providers/devices.service.ts

```diff
@@ -19,6 +19,9 @@ export class DevicesService {
   public getDevices = (): Device[] => {
     return this.devices;
   }
+  public setDevices = (dev: Device[]): void => {
```

Comment:
That is not good testing etiquette.
I can explain it in detail if you want, but running `newDevice` several times is not an option